smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/retries/StandardRetryIntegration.kt

```diff
@@ -0,0 +1,79 @@
+import software.amazon.smithy.kotlin.codegen.model.namespace
+import software.amazon.smithy.kotlin.codegen.rendering.ClientConfigProperty
+import software.amazon.smithy.kotlin.codegen.rendering.protocol.HttpFeatureMiddleware
+import software.amazon.smithy.kotlin.codegen.rendering.protocol.ProtocolGenerator
+import software.amazon.smithy.kotlin.codegen.rendering.protocol.ProtocolMiddleware
+
+object RetryConfigSection : SectionId
+
+/**
+ * Adds retry wrappers around operation invocations, driven by client config (which specifies the [RetryStrategy]) and
+ * a middleware (which specifies the [RetryPolicy][aws.smithy.kotlin.runtime.retries.RetryPolicy]). This class is
+ * `open` and may be subclassed to customize the strategy and/or policy. Note that codegen which specifies a subclass of
+ * this should probably configure [replacesIntegrations] to remove the base implementation—otherwise, multiple copies of
+ * the config properties may be included in client config.
+ */
+class StandardRetryIntegration : KotlinIntegration {
+    override fun additionalServiceConfigProps(ctx: CodegenContext): List<ClientConfigProperty> =
+        listOf(
+            ClientConfigProperty {
+                val retryStrategyBlock = """
+                    run {
+                        val strategyOptions = StandardRetryStrategyOptions.Default
+                        val tokenBucket = StandardRetryTokenBucket(StandardRetryTokenBucketOptions.Default)
+                        val delayer = ExponentialBackoffWithJitter(ExponentialBackoffWithJitterOptions.Default)
+                        StandardRetryStrategy(strategyOptions, tokenBucket, delayer)
+                    }
+                """.trimIndent()
+
+                symbol = buildSymbol {
+                    name = "RetryStrategy"
+                    namespace(KotlinDependency.CORE, "retries")
+                    nullable = false
+                    reference(RuntimeTypes.Core.Retries.Impl.StandardRetryStrategy, ContextOption.USE)
+                    reference(RuntimeTypes.Core.Retries.Impl.StandardRetryStrategyOptions, ContextOption.USE)
+                    reference(RuntimeTypes.Core.Retries.Impl.StandardRetryTokenBucket, ContextOption.USE)
+                    reference(RuntimeTypes.Core.Retries.Impl.StandardRetryTokenBucketOptions, ContextOption.USE)
+                    reference(RuntimeTypes.Core.Retries.Impl.ExponentialBackoffWithJitter, ContextOption.USE)
+                    reference(RuntimeTypes.Core.Retries.Impl.ExponentialBackoffWithJitterOptions, ContextOption.USE)
+                }
+                name = "retryStrategy"
+                documentation = """
+                    The [RetryStrategy] implementation to use for service calls. All API calls will be wrapped by the
+                    strategy.
+                """.trimIndent()
+                constantValue = retryStrategyBlock
+            }
+        )
+
+    override fun customizeMiddleware(
```

Comment:
correctness

Should probably only customize the middleware if it's an HTTP protocol. Another reason to maybe move this to a set of [default HTTP middleware](https://github.com/awslabs/smithy-kotlin/blob/45ff27e749f7454b232d2846b96ec2c8c40d48fa/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpBindingProtocolGenerator.kt#L63) instead of an integration.